src/ripple/rpc/impl/ShardArchiveHandler.cpp

```diff
@@ -61,24 +94,156 @@ ShardArchiveHandler::hasInstance()
     return instance_.get() != nullptr;
 }
 
-ShardArchiveHandler::ShardArchiveHandler(Application& app, bool validate)
+ShardArchiveHandler::ShardArchiveHandler(WithoutRecovery const& tag,
+                                         Application& app, bool validate)
     : app_(app)
-    , downloadDir_(get(app_.config().section(
-        ConfigSection::shardDatabase()), "download_path",
-            get(app_.config().section(ConfigSection::shardDatabase()),
-                "path", "") + "/download"))
+    , j_(app.journal("ShardArchiveHandler"))
+    , downloadDir_(getDownloadDirectory(app.config()))
+    , stateDir_(getStateDirectory(app.config()))
     , validate_(validate)
     , timer_(app_.getIOService())
     , process_(false)
-    , j_(app.journal("ShardArchiveHandler"))
 {
     assert(app_.getShardStore());
+
+    auto setup = setup_DatabaseCon(app_.config());
+    setup.dataDir = stateDir_;
+
+    create_directory(stateDir_);
+
+    conn_ = std::make_unique<DatabaseCon>(
+        stateDir_,
+        "stateDatabase",
+        DatabasePragma,
+        DatabaseInit);
+
+    auto db = conn_->checkoutDb();
```

Comment:
No need for a LockedSociSession. conn_ is always used while holding the lock so getSession() can be used instead. Consider renaming the variable to 'session'.

eg.
```
auto& session {conn_->getSession()};
```

Reply:
Nit: Consider also renaming conn_ to something like sqliteDB or dbConn.

Reply:
Done